Crashing every eight hours

**What happened**:
New DNSSEC ```periodicClean``` from #4537 panics.
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Perform one or more DNSSEC queries then wait up to 8 hrs for the panic.
**Anything else we need to know?**:
PR incoming
**Environment**:

- the version of CoreDNS: Master (8487111)
- Corefile: 
```
example.org {
    dnssec {
        key file Kexample.org.+013+45330
    }
    whoami
}
```
- logs, if applicable:
```
CoreDNS-1.8.3
linux/amd64, go1.16.3, 84871118-dirty
panic: interface conversion: interface {} is []dns.RR, not *dns.RRSIG
```

--- a/plugin/dnssec/cache.go
+++ b/plugin/dnssec/cache.go
@@ -33,9 +33,10 @@ func periodicClean(c *cache.Cache, stop <-chan struct{}) {
 			// the signature
 			is75 := time.Now().UTC().Add(sixDays)
 			c.Walk(func(items map[uint64]interface{}, key uint64) bool {
-				sig := items[key].(*dns.RRSIG)
-				if !sig.ValidityPeriod(is75) {
-					delete(items, key)
+				for _, rr := range items[key].([]dns.RR) {
+					if !rr.(*dns.RRSIG).ValidityPeriod(is75) {
+						delete(items, key)
+					}
 				}
 				return true
 			})

